interface/wx/httprequest.h

```diff
@@ -0,0 +1,232 @@
+        }
+    @endcode
+
+    System wide configuration like proxy and SSL certificates will be used
+    when possible.
+
+    @section descriptions Implementation Descriptions
+
+    The following APIs are used per platform, additional documentation
+    about supported features may be found in ther documentation
+
+    @par WinHTTP on Windows
+
+    This API is available on all supported Windows Versions and supports
+    HTTP and HTTPs up to HTTP/2.
+
+    <a href="https://docs.microsoft.com/en-us/windows/desktop/WinHttp/about-winhttp">
+    API Documentation</a>
+
+    @par NSURLSession on macOS and iOS
+
+    This API is available since macOS 10.9 and iOS 7.0.
+
+    <a href="https://developer.apple.com/documentation/foundation/urlsession">
+    API Documentation</a>
+
+    @par libcurl on Linux
+
+    This library is available on most Unix distributions.
+
+    <a href="https://curl.haxx.se/libcurl/">
+    API Documentation</a>
+
+    @since 3.1.2
+    @library{wxnet}
+    @category{net}
+
+    @see wxHTTPSession
+*/
+class wxHTTPRequest
+{
+public:
+    /**
+        Create a new request instance
+
+        @param url The URL of the HTTP resource for this request
+        @param session If no wxHTTPSession is give the default session will
+                       be used
+
+        @since 3.1.2
+    */
+    wxHTTPRequest(const wxString& url, wxHTTPSession session = NULL);
+
+    /**
+
+        @since 3.1.2
+    */
+    ~wxHTTPRequest();
+
+    /**
+        Adds a request header send by this request.
+
+        @param name Name if the header
+        @param value String value of the header
+
+        @since 3.1.2
+    */
+    void AddRequestHeader(const wxString& name, const wxString& value);
+
+    /**
+        Set HTTP method.
+
+        Set <a href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html">common</a>
+        or expanded HTTP method.
+
+        Overrides GET or POST methods that is used by default.
+
+        @param method
+            HTTP method name, e.g. "GET".
+
+        @since 3.1.2
+    */
+    void SetRequestMethod(const wxString& method);
+
+    /**
+        Set the text to be posted to the server.
+
+        After a successful call to this method, the request will use HTTP @c
+        POST instead of the default @c GET when it's executed.
+
+        @param data
+            The data to post.
+        @param contentType
+            The value of HTTP "Content-Type" header, e.g. "text/html;
+            charset=UTF-8".
+
+        @since 3.1.2
+    */
+    void SetRequestData(const wxString& data, const wxString& contentType);
+
+    /**
+        Set the binary data to be posted to the server.
+
+        The next request will
+        be an HTTP @c POST instead of the default HTTP @c GET and the given @a
+        data will be posted as the body of this request.
+
+        @since 3.1.2
+    */
+    void SetRequestData(const wxInputStream*, const wxString& contentType);
```

Comment:
Why not use `void SetRequestData(const wxMemoryBuffer& buffer, const wxString& contentType);`
Seems more simple to me

Reply:
You shouldn't have to make an extra copy of the data in memory if you already have a stream providing it.

Reply:
I'm especially thinking about a something common like uploading a file (with a multipart mime form data stream, which would of course would need to be implemented first)